### ISSUE TITLE
Helm Chart Deployment Option

### ISSUE THREAD

**Issue by tlchaffi** [PacoVK/tapir#290]
Feature request.
It'd be excellent if we could reference a Helm Chart, from the Helm Chart registry. That way we could use the Terraform Hashicorp / Helm provider to deploy Tapir into a kubernetes service.

**Comment by PacoVK** [comment]
Good idea, feel free to raise a PR. I could look into that as well but currently have some time constraints.

**Comment by jonasz-lasut** [comment]
@PacoVK if you need help I can implement a helm chart for this repo. It looks like a really simple deployment from kubernetes point of view so it shouldn't take that much time.
If you're interested please let me know in the comment, I'll then contact you on LinkedIn or Mastodon and we can talk about the details :)

**Comment by tlchaffi** [comment]
Sign me up too. I've deployed several Helm charts to k8 using ArgoCD. But I've never made a Helm chart. I'd love to help.

**Comment by jonasz-lasut** [comment]
@tlchaffi if Paco is ok with that then it works for me too. I've developed some golden hełm charts for different companies before but it's going to be faster in a team 😁

**Comment by PacoVK** [comment]
Thanks, really appreciate that you are willing to take care! Feel free to reach out to share ideas or let us have a discussion here, that anybody can chime in. 👍

**Comment by tlchaffi** [comment]
For the storage layer of Tapir, for example, on Azure, it refers to an Azure storage account / blob account.

Do we want the Helm chart to resolve the storage layer? Or should it do like a k8 mounted disk?

**Comment by jonasz-lasut** [comment]
That's on the service account level, you have workload identity using AWS/GCP/whatever annotation (`"eks.amazonaws.com/role-arn" = aws_iam_role.tapir.arn` in terraform). I believe we should be still doing workload identity (sorry if that's not the name used on AWS, I'm a GCP guy at heart) with annotations, just leave default as empty annotation object `{}` instead of AWS role-arn.

However there's also ClusterRole and binding to that SA that allows get/watch/list on services, ingresses and nodes. I wonder if those privileges are needed for Tapir to function properly or not.

**Comment by PacoVK** [comment]
@jonasz-lasut concerning the ClusterRole i need to double check again, most of the code for K8s in the example was based on #253 but i guess this is not mandatory for Tapir to function at least in a K8s context. There may be an exception for EKS, but thats open to check. AFAIK AWS also introduced an easier way to handle Pod permissions at reinvent 23

**Comment by jonasz-lasut** [comment]
@PacoVK ok, for now I've left it behind createClusterRoles flag

You can find initial draft of the chart under [Chart](https://github.com/jonasz-lasut/tapir/tree/issue-290/helm-chart-deployment/charts) for now.
I'll be working on CI pipeline for it tomorrow and I'll do some cleanup in the values and add documentation to them.  I'll setup helm docs and changelog generation on the workflow

**Comment by jonasz-lasut** [comment]
Hi, I've opened a pull request with full helm chart with Helm-docs annotations, generated helm docs and script to generate them locally and a basic pipeline which lints helm chart and checks if the docs were generated if any changes are made to the helm chart.
https://github.com/PacoVK/tapir/pull/375

I didn;t iplemented bumping up AppVersion in the helm chart from the build/deploy pipeline of `tapir` as I didn't want to break anything and wanted to have separation between helm chart and tapir. I can open another PR later :)

**Comment by PacoVK** [comment]
I think for the first step this is more than enough to just mention the version bump in the [release notes](https://github.com/PacoVK/tapir/blob/main/dev/docs/release.md) We could automate separately then :)

**Comment by PacoVK** [comment]
@all-contributors please add @jonasz-lasut for example, code

**Comment by PacoVK** [comment]
@all-contributors please add @tlchaffi for review

**Comment by dennismdejong** [comment]
Thanks for the request! A few questions:

1. **Existing helm chart**: There is already a Helm chart at https://github.com/PacoVK/tapir-helm. Would it be sufficient to improve/reference that chart, or are you looking for something different?

2. **Scope**: Are you looking to have Tapir *host* Helm charts (like a private Helm repo), or to provide a Helm chart to *deploy* Tapir itself? The issue description suggests the latter — just checking.

3. **Integration**: Would you need this as a new section in the Tapir UI for managing the Helm deployment, or just have the chart available as a deployment option?
